pennylane/templates/subroutines/qsvt.py

```diff
@@ -321,17 +321,20 @@ def compute_decomposition(
         UA_adj = copy.copy(UA)
 
         for idx, op in enumerate(projectors[:-1]):
-            qml.apply(op)
+            if qml.QueuingManager.recording():
+                qml.apply(op)
             op_list.append(op)
 
             if idx % 2 == 0:
-                qml.apply(UA)
+                if qml.QueuingManager.recording():
+                    qml.apply(UA)
                 op_list.append(UA)
 
             else:
                 op_list.append(adjoint(UA_adj))
 
-        qml.apply(projectors[-1])
+        if qml.QueuingManager.recording():
+            qml.apply(projectors[-1])
         op_list.append(projectors[-1])
```

Comment:
Will this need to become a rule of thumb for all template implementations?

Reply:
It already was.

We just didn't catch that it because there are no unit tests for `QSVT.compute_decomposition`.